### PR TITLE
Radio Update - Additional Classes added, Documentation in progress, Tests in progress

### PR DESCRIPTION
**Pull request by cynthiaxsa**
Pushing my current progress in the Radio Feature. Many aspects of it are uncompleted, though hopefully this gives everyone a better idea of what I am currently working towards. In this patch, I have done a few of the following:

- Added documentation and TODOs throughout 
- Readjusted the structure of my classes, added several needed additional classes especially along the IA level. 
- Completed the Radiosearch use case for the search engine - yet to be tested. 
- Set up the skeleton of the tests that I am planning on implementing in the future. 

A note! At the stage it is currently at, Radio is not heavily involved or otherwise dependent on other features. There should not be many issues or concerns needed on that front, but if there are any do let me know!

### CODE REVIEW

File: src/main/java/abr/radio_abr/RadioStationInputBoundary.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package abr.radio_abr;
+
+import interface_adaptors.radio_ia.RadioPlayerViewModel;
```
Comment (Starliyiyu) [inline src/main/java/abr/radio_abr/RadioStationInputBoundary.java:3]: abr depending on ia

**Review by varygx — CHANGES_REQUESTED** [review]
Please pull from main and merge the changes into your branch. There's a lot of files unrelated to your PR that are being changed. For example, I'm seeing a test class being reverted to a very old version.

File: src/main/java/abr/search_engine_abr/SearchUserAbr.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package abr.search_engine_abr;
 
-import ds.user_search_engine.SearchEngineFileGateway;
+import abr.user_avatar_image_management_abr.ds.user_search_engine.SearchEngineFileGateway;
```
Comment (Starliyiyu) [inline src/main/java/abr/search_engine_abr/SearchUserAbr.java:3]: there would be package arrangement problems if this line runs no problem; if you are trying to import search engine gateways, import  SearchEngineDatabaseGateway instead, the filegateway is a class in ia & violates clean architecture

**Review by Starliyiyu — APPROVED** [review]
some import statements seem weird, not sure if this is because your local repo didn't update to the latest version, everything else looks good

File: src/main/java/framework/user_screen/UserLogUI.java [hunk 3]
```diff
@@ -117,7 +101,17 @@ private void createScreenComponents() {
...
+    // Temporary test file
+    public static void main(String[] args) {
+        UserLoginDataBaseGateway dataBaseGateway = new UserLoginFileGateway();
+        UserLogViewModel viewModel = new UserLogViewModel();
+        UserStatusViewModel statusViewModel = new UserStatusViewModel();
+        UserLogOutputBoundary presenter = new UserLogPresenter(viewModel, statusViewModel);
+        UserLogInputBoundary useCase = new UserLogUseCase(presenter, dataBaseGateway);
+        UserLogController controller = new UserLogController(useCase);
+        new UserLogUI(controller, viewModel);
     }
```
Comment (2368883029) [inline src/main/java/framework/user_screen/UserLogUI.java:115]: This file is the old version of the UserLogUI, please preserve the newer version in Main Branch

File: src/main/java/abr/user_avatar_image_management_abr/ds/user_database/UserFileReader.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package ds.user_database;
+package abr.user_avatar_image_management_abr.ds.user_database;
```
Comment (2368883029) [inline src/main/java/abr/user_avatar_image_management_abr/ds/user_database/UserFileReader.java:1]: There seems to be a FileArrangement problem, ds File is now under ABR.

**Review by 2368883029 — CHANGES_REQUESTED** [review]
You should solve these problems by 
1) move .ds directory out of abr.userAvatarManagement directory
2) pull changes from the main, and preserve UserLogUI from the Main manually by resolving conflicts.

File: src/main/java/abr/radio_abr/RadioStationInputBoundary.java (hunk 1 above)
Comment (cynthiaxsa) [reply src/main/java/abr/radio_abr/RadioStationInputBoundary.java:3]: Gotcha, I've adjusted it now!

**Comment by cynthiaxsa** [comment]
With assistance from Allen I've been able to update my branch properly, so it should now have all files up to date. I'm going to re-push now!

**Comment by cynthiaxsa** [comment]
Closing for now as I figure this out

**Pull request closed without merging**